### PR TITLE
Display no authenticators error

### LINKED ISSUE #51: Display server side errors in form
When some kind of error occurs on the server side, `django-fido` simply displays `An unknown error has occurred.` message.

However, server often sends detailed message about what happened. E.g. when user has no authenticators, server returns status code 404 with response:

```
{"error": "Can't create FIDO 2 authentication request, no authenticators."}
```

This message would be much more helpful for the user. It's necessary to handle the translations as well.

### PR DESCRIPTION
**Pull request by stinovlas**
Closes #51

### CODE REVIEW

File: django_fido/views.py [hunk 1]
```diff
@@ -99,7 +108,11 @@ def get(self, request: HttpRequest) -> JsonResponse:
         try:
             request_data, state = self.create_fido2_request()
         except ValueError as error:
-            return JsonResponse({'error': force_text(error)}, status=BAD_REQUEST)
+            return JsonResponse({
+                'error_code': getattr(error, 'error_code', None),
+                'message': force_text(error),
+                'error': force_text(error),
```
Comment (tpazderka) [inline django_fido/views.py]: I would at least add a note here that the `error` is deprecated.

Also a note in CHANGELOG should be added.

File: django_fido/js/fido2.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ function createTranslations() {
         'ConstraintError': gettext('The security token (FIDO 2) does not meet required criteria.'),
         'NotAllowedError': gettext('The request has been refused either by you, your browser, or your'
                                    + ' operating system. Or the request timed out.'),
+        'NoAuthenticatorsError': gettext('Can\'t create FIDO 2 authentication request, no authenticators.'),
```
Comment (ziima) [inline django_fido/js/fido2.js]: How about `...no authenticators found`?

File: django_fido/views.py [hunk 3]
```diff
@@ -99,7 +108,11 @@ def get(self, request: HttpRequest) -> JsonResponse:
         try:
             request_data, state = self.create_fido2_request()
         except ValueError as error:
-            return JsonResponse({'error': force_text(error)}, status=BAD_REQUEST)
+            return JsonResponse({
+                'error_code': getattr(error, 'error_code', None),
```
Comment (ziima) [inline django_fido/views.py]: I'd set `Fido2ServerError` as a default `error_code`. At least to split errors from server and client.

File: django_fido/views.py [hunk 4]
```diff
@@ -263,7 +276,8 @@ def create_fido2_request(self) -> Tuple[Dict, Dict]:
         assert user and user.is_authenticated, "User must not be anonymous for FIDO 2 requests."
         credentials = self.get_credentials(user)
         if not credentials:
-            raise ValueError("Can't create FIDO 2 authentication request, no authenticators.")
+            raise Fido2Error("Can't create FIDO 2 authentication request, no authenticators.",
+                             error_code='NoAuthenticatorsError')
```
Comment (ziima) [inline django_fido/views.py]: Error codes should be defined in an enum.

File: django_fido/views.py (hunk 1 above)
Comment (stinovlas) [reply django_fido/views.py]: Fixed.

File: django_fido/js/fido2.js (hunk 2 above)
Comment (stinovlas) [reply django_fido/js/fido2.js]: Fixed.

File: django_fido/views.py (hunk 3 above)
Comment (stinovlas) [reply django_fido/views.py]: Fixed.

File: django_fido/views.py (hunk 4 above)
Comment (stinovlas) [reply django_fido/views.py]: Fixed.

**Pull request merged**